import grpc
import phrases_service_pb2
import phrases_service_pb2_grpc

channel = grpc.insecure_channel('81.94.150.171:10049')
stub = phrases_service_pb2_grpc.GTAGameServiceStub(channel)

phrases = [
    ("Experience has taught me that a man like you can be very loyal for normal money, but when they unite, men become too greedy.", "GTA 3"),
    ("Hey, this is my favorite cleaner guy! I'm proud of you. You beat the crap out of those fat ball shakers.", "GTA 3"),
    ("I already told you, man, dont tease me! My taxes pay your salary, you fagot!", "GTA 3"),
    ("Sir, this is a commercial radio station owned by Love Media. I get paid from advertising revenue.", "GTA 3"),
    ("I'm going to get the gun. Guns don't kill people. People kill people. Although weapons help them with this", "GTA 3"),
    ("I am an old lady, for Christs sake!", "GTA 3"),
    ("When a problem arises, a fool turns his back on it. While the sage meets her face", "GTA 3"),
    ("Nothing lowers property prices like a good old-fashioned showdown. Unless there's an outbreak of plague, but in that case we'll go too far", "GTA 3"),
    ("Sorry baby, I am an ambitious girl and you dont have much time", "GTA 3"),
    ("This is my money, you should think about it every second", "GTA Vice City"),
    ("A little tighter between the legs - both for me and for me too", "GTA Vice City"),
    ("I have been looking for such a bird for centuries", "GTA Vice City"),
    ("There are barbarians at the gates", "GTA Vice City"),
    ("Someday I'll save your ass and I think you'll want to kiss mine in return", "GTA Vice City"),
    ("Sorry, Tommy! This is Vice City! It's just business , nothing personal! ", "GTA Vice City"),
    ("Nobody does what they want. They just do what you let them do", "GTA Vice City"),
    ("So what was the plan, Tommy?‚Äù You decided that I would fall for it, take counterfeit money and run away with my tail between my legs?!", "GTA Vice City"),
    ("Why is it always like this with me?! As soon as you stick your head out of the ditch, life immediately throws a portion of crap in your face!! ", "GTA Vice City"),
    ("Tommy! Tommy - how many years, how many winters! 15 years and it seems like just yesterday! ", "GTA Vice City"),
    ("I trusted you, Tommy! I wanted to make you...", "GTA Vice City"),
    ("Let me guess you think I need a guardian angel", "GTA Vice City"),
    ("We just go into the bank, wave the guns, take the money and leave. Like in the movies", "GTA Vice City"),
    ("Grove Street. House. At least, he was until I messed things up here", "GTA San Andreas"),
    ("AH SHIT, HERE WE GO AGAIN", "GTA San Andreas"),
    ("Are you a professional idiot or a talented amateur?!", "GTA San Andreas"),
    ("You know what, Carl, you're a fucking idiot. All your life you wanted something for nothing, now you got something and dont know what to do with it", "GTA San Andreas"),
    ("Dude, if you can eat your snack while everyone else is losing it, you're pretty cool", "GTA San Andreas"),
    ("Hey little man, why do you act like this is your woman, huh? She's with me, donkey's ass, so go to the star, I'll take care of her", "GTA San Andreas"),
    ("He burst into the house - shaggy, unkempt,I drank furniture cleaner, howling,And then I ate apricot pie,Tapestry Cockerels and aloe", "GTA San Andreas"),
    ("I'd like two No. 9s, No. 9 - a large portion, No. 6, with extra sauce, No. 7, two No. 45s, one of them with cheese, and a large soda", "GTA San Andreas"),
    ("I know you're blind, but you gotta see this", "GTA San Andreas"),
    ("Carl, you motherfucking piece of shit gang-banging cocksucker!", "GTA San Andreas"),
    ("We can shit on you from such a height, youd think god himself has crapped on you", "GTA San Andreas"),
    ("That shit was tight! Tight? Man, that shit was shit", "GTA San Andreas"),
    ("Like the book says...we are both blessed and cursed.", "GTA San Andreas"),
    ("Who the hell are you? Where is my old driver?! Unlock the damn door! I cant fucking swim, you darned psychopath!", "GTA San Andreas"),
    ("The same place where I buy my pants. This is America!", "GTA San Andreas"),
    ("People who try to fuck me then take a long time to heal their ass!", "GTA 4"),
    ("People are imperfect... Life is a mess", "GTA 4"),
    ("In war, the experienced and seasoned fool the young and stupid, forcing them to kill each other", "GTA 4"),
    ("What the eyes cannot see, the heart will feel", "GTA 4"),
    ("Capitalism is a dirty business!", "GTA 4"),
    ("We all do stupid things-that's the hallmark of being human", "GTA 4"),
    ("Listen... What's done is done. One mistake doesn't make me a devil", "GTA 4"),
    ("My dreams won't come true because I don't have them", "GTA 4"),
    ("It's not boring to even watch the paint dry! ", "GTA 4"),
    ("Before you can learn to fly, you need to learn to crawl", "GTA 4"),
    ("Life is complicated. I killed people, smuggled people, sold people. Perhaps things will change here", "GTA 4"),
    ("Not only is Ireland green, but so is money!", "GTA 4"),
    ("Why do you need to eat cabbage if you have potatoes?", "GTA 4"),
    ("You can't shake anyone's hand if your fingers are clenched into a fist", "GTA 4"),
    ("Not everything in this life can be changed. You must have the courage to change what can be changed and the wisdom to accept what cannot be changed", "GTA 4"),
    ("You begin to understand what's what. You can't take interest from a dead person", "GTA 4"),
    ("For a collective farmer, you are not that stupid!", "GTA 4"),
    ("I sleep with women, beat men! Not the other way around!", "GTA 4"),
    ("Mr. Raspberry died a brave death, and in the process brought great pleasure to a lonely man", "GTA 5"),
    ("Every day you forget a thousand little things. Let this be one of them", "GTA 5"),
    ("Half the world is starving, but your Bobik eats only organic products", "GTA 5"),
    ("Yes, compared to you, I am the most normal bastard I know", "GTA 5"),
    ("You became cool because you followed the rules, and now you think you're cooler than the rules, but there will always be someone who will throw you into the shit", "GTA 5"),
    ("Tell me exactly what you want, and I will clearly explain to you why it is impossible!", "GTA 5"),
    ("Did you tell me this on the phone? Fucking Ron! Have you gone crazy?!", "GTA 5"),
    ("Hey! If anyone will kill you, my friend, it will be me! ", "GTA 5"),
    ("I will find this goat who pissed on me and eat his liver!", "GTA 5"),
    ("Weapons and prohibitions in this area only go through Trevor Phillips Enterprises. Or they dont pass at all! ", "GTA 5"),
    ("Remember: there is only mass, fluidity and incompressibility", "GTA 5"),
]

def register_user():
    request = phrases_service_pb2.RegisterRequest(username="old3gg", password="old3gg")
    response = stub.RegisterUser(request)
    print("RegisterUser Response:", response)

def login_user():
    request = phrases_service_pb2.LoginRequest(username="old3gg", password="old3gg")
    response = stub.LoginUser(request)
    print("LoginUser Response:", response)
    return response.authToken

def get_phrase(auth_token):
    request = phrases_service_pb2.GetPhraseRequest(authToken=auth_token)
    response = stub.GetPhrase(request)
    print("GetPhrase Response:", response)
    return response.text

def check_answer(auth_token, phrase, answer):
    request = phrases_service_pb2.CheckAnswerRequest(authToken=auth_token, text=phrase, answer=answer)
    print(request)
    response = stub.CheckAnswer(request)
    print("CheckAnswer Response:", response)
    return response.correct, response.correctCount

def find_correct_answer(phrase):
    for p, answer in phrases:
        if p == phrase:
            return answer
    return None

def main():
    register_user()
    
    auth_token = login_user()
    
    for _ in range(100):
        phrase = get_phrase(auth_token)
        print("Received Phrase:", phrase)
        correct_answer = find_correct_answer(phrase)
        if correct_answer:
            correct, correct_count = check_answer(auth_token, phrase, correct_answer)
            print(f"Correct: {correct}, CorrectCount: {correct_count}")
        else:
            print("Phrase not found in the dictionary!")

if __name__ == "__main__":
    main()
